buildpress-config/src/main/scala/buildpress/config/Config.scala

```diff
@@ -0,0 +1,79 @@
+package buildpress.config
+
+import java.net.URI
+import java.nio.charset.StandardCharsets
+import java.nio.file.{Files, Path, Paths}
+import scala.util.Try
+import io.circe.generic.JsonCodec
+import io.circe.{Decoder, Encoder, Printer}
+
+object Config {
+  val BuildpressCacheFileName = "buildpress-repo-cache.json"
+
+  implicit val pathEnc: Encoder[Path] =
+    Encoder.encodeString.contramap[Path](_.toString)
+  implicit val pathDec: Decoder[Path] =
+    Decoder.decodeString.emapTry(s => Try(Paths.get(s)))
+
+  implicit val uriEnc: Encoder[URI] =
+    Encoder.encodeString.contramap[URI](_.toString)
+  implicit val uriDec: Decoder[URI] =
+    Decoder.decodeString.emapTry(s => Try(URI.create(s)))
+
+  @JsonCodec
```

Comment:
We would usually avoid using these macro annotations at all costs and instead use semi-automatic derivation manually. However, i don't think it's a big deal so we can live with this code for now :smile: